cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go

```diff
@@ -68,13 +68,13 @@ func NewNodeDeletionBatcher(ctx *context.AutoscalingContext, csr *clusterstate.C
 }
 
 // AddNode adds node to delete candidates and schedule deletion.
-func (d *NodeDeletionBatcher) AddNode(node *apiv1.Node, drain bool) error {
+func (d *NodeDeletionBatcher) AddNode(node *apiv1.Node, nodeGroupId string, drain bool) error {
 	// If delete interval is 0, than instantly start node deletion.
 	if d.deleteInterval == 0 {
 		nodeGroup, err := deleteNodesFromCloudProvider(d.ctx, []*apiv1.Node{node})
 		if err != nil {
 			result := status.NodeDeleteResult{ResultType: status.NodeDeleteErrorFailedToDelete, Err: err}
-			CleanUpAndRecordFailedScaleDownEvent(d.ctx, node, nodeGroup.Id(), drain, d.nodeDeletionTracker, "", result)
+			CleanUpAndRecordFailedScaleDownEvent(d.ctx, node, nodeGroupId, drain, d.nodeDeletionTracker, "", result)
```

Comment:
One thing of note is that it looks like we call `CleanUpAndRecordFailedScaleDownEvent` both here _and_ here: https://github.com/kubernetes/autoscaler/blob/265e57c1636298729002695326db32205d866f48/cluster-autoscaler/core/scaledown/actuation/actuator.go#L367

Is that intended? Shouldn't it only be called one?

Reply:
@yaroslava-serdiuk curious if you have thoughts here since you authored these bits

Reply:
Ya you're right, should have read through a bit more closely, thanks for clearing that up!

Reply:
No problem :D